registry/endpoint.go

```diff
@@ -100,15 +99,18 @@ func newEndpoint(address string, secure bool) (*Endpoint, error) {
 		return nil, err
 	}
 	endpoint.IsSecure = secure
+	tr := NewTransport(ConnectTimeout, endpoint.IsSecure)
+	endpoint.client = HTTPClient(transport.NewTransport(tr, DockerHeaders(metaHeaders)...))
```

Comment:
I see. It seems like we need to get control over client instantiation. We need to differentiate between headers that are always present (Auth, ident, etc.) from those that are part of the specific request flow.

I'm not sure how much this should affect this particular PR.